src/renderer/src/components/song/song-item/SongItem.tsx

```diff
@@ -57,32 +51,59 @@ const SongItem: Component<SongItemProps> = ({
   });
 
   return (
-    <div
-      class="group relative isolate select-none rounded-md"
-      classList={{
-        "outline outline-2 outline-accent": selectedSong().path === song.path,
-      }}
-      data-active={selectedSong().path === song.path}
-      ref={item}
-      data-url={song.bg}
-      onContextMenu={showMenu}
+    <Popover
+      isOpen={localShow}
+      onValueChange={setLocalShow}
+      middlewares={[flip(), offset({ crossAxis: 30 })]}
+      placement="right"
+      offset={15}
     >
-      <SongImage
-        class="absolute inset-0 z-[-1] h-full w-full rounded-md bg-cover bg-center bg-no-repeat opacity-30 group-hover:opacity-90"
+      <Portal>
+        <Popover.Overlay />
+        <Popover.Content
+          onClick={(e) => {
+            e.stopImmediatePropagation();
+            setLocalShow(false);
+          }}
+        >
+          {...children}
+        </Popover.Content>
+      </Portal>
+      <div
+        class="group relative isolate z-20 select-none rounded-md"
         classList={{
-          "opacity-90": selectedSong().path === song.path,
+          "outline outline-2 outline-accent": selectedSong().path === song.path,
         }}
-        src={song.bg}
-        group={group}
-      />
+        data-active={selectedSong().path === song.path}
+        ref={item}
+        data-url={song.bg}
+        onContextMenu={() => setLocalShow(true)}
+      >
+        <SongImage
+          class="absolute inset-0 z-[-1] h-full w-full rounded-md bg-cover bg-center bg-no-repeat opacity-30 group-hover:opacity-90"
+          classList={{
+            "opacity-90": selectedSong().path === song.path,
+          }}
+          src={song.bg}
+          group={group}
+        />
+
+        <div class="flex flex-row justify-between rounded-md bg-black/50">
+          <div class="z-20 flex min-h-[72px] flex-col justify-center overflow-hidden rounded-md p-3">
+            <h3 class="text-shadow text-[22px] font-extrabold leading-7 shadow-black/60">
+              {song.title}
+            </h3>
+            <p class="text-base text-subtext">{song.artist}</p>
+          </div>
 
-      <div class="flex min-h-[72px] flex-col justify-center overflow-hidden rounded-md bg-black/50 p-3">
-        <h3 class="text-shadow text-[22px] font-extrabold leading-7 shadow-black/60">
-          {song.title}
-        </h3>
-        <p class="text-base text-subtext">{song.artist}</p>
+          <Popover.Trigger class="opacity-0 transition-opacity group-hover:opacity-100">
+            <Button variant={"ghost"} size={"icon"} class="z-50 mr-1 rounded-lg">
```

Comment:
The `Popover.Trigger` is already a button, no need to put another button inside, replace it with something like:

``` diff
-  <Button variant={"ghost"} size={"icon"} class="z-50 mr-1 rounded-lg">
+ <div class="mr-2 grid aspect-square size-9 place-items-center rounded border-solid border-stroke bg-transparent p-1 text-text hover:bg-surface">
```